src/hobbits-plugins/operators/Edit/edit.cpp

```diff
@@ -0,0 +1,236 @@
+#include "edit.h"
+#include "editeditor.h"
+#include "bitarray.h"
+
+Edit::Edit()
+{
+    QList<ParameterDelegate::ParameterInfo> infos = {
+        // TODO: add parameters like {"myparametername", ParameterDelegate::ParameterType::Decimal}
+        {"start", ParameterDelegate::ParameterType::Decimal},
+        {"length", ParameterDelegate::ParameterType::Decimal},
+        {"new_bits_in_range", ParameterDelegate::ParameterType::String, false},
+        {"edit_type", ParameterDelegate::ParameterType::String}
+    };
+
+    m_delegate = ParameterDelegate::create(
+                    infos,
+                    [this](const Parameters &parameters) {
+                        // TODO: use parameters to describe action better
+                        return QString("Apply %1").arg(this->name());
+                    },
+                    [](QSharedPointer<ParameterDelegate> delegate, QSize size) {
+                        Q_UNUSED(size)
+                        return new EditEditor(delegate);
+                    });
+}
+
+OperatorInterface* Edit::createDefaultOperator()
+{
+    return new Edit();
+}
+
+QString Edit::name()
+{
+    return "Edit";
+}
+
+QString Edit::description()
+{
+    // TODO: create actual description
+    return "Load bits, and modify them individually.";
+}
+
+QStringList Edit::tags()
+{
+    // TODO: add relevant tags
+    return {"Generic"};
+}
+
+QSharedPointer<ParameterDelegate>  Edit::parameterDelegate()
+{
+    return m_delegate;
+}
+
+int Edit::getMinInputContainers(const Parameters &parameters)
+{
+    Q_UNUSED(parameters)
+    return 1;
+}
+
+int Edit::getMaxInputContainers(const Parameters &parameters)
+{
+    Q_UNUSED(parameters)
+    return 1;
+}
+
+int Edit::getUnitSize(Parameters parameters) {
+    if (parameters.value("edit_type").toString() == "hex") {
+        return 4;
+    }
+    else if (parameters.value("edit_type").toString() == "ascii") {
+        return 8;
+    }
+    else {
+        return 1;
+    }
+}
+
+QSharedPointer<BitArray> Edit::parseBits(QString newBitsInRange) 
```

Comment:
I think these `parseX` functions could be better as a single function because they share a lot of similarities. Why not just a `QSharedPointer<BitArray> parseNewBits(Parameters &params)` function? That will avoid complexity in `operateOnBits` and consolidate the parsing code.